src/index.test.ts

```diff
@@ -63,6 +63,147 @@ describe('basic go usage', () => {
+          res(2);
+        }, 10)
+      ),
+    errorFn: () => new Promise((_res, rej) => setTimeout(() => rej(new Error('Computer says no')), 10)),
+  };
+
+  it('resolves successful asynchronous functions within the timout limit', async () => {
+    const res = await timeoutGo(operations.successFn, { timeoutMs: 20 });
+    expect(res).toEqual(success(2));
+  });
+
+  it('resolves unsuccessful asynchronous functions within the timout limit', async () => {
+    const res = await timeoutGo(operations.errorFn, { timeoutMs: 20 });
+    expect(res).toEqual(fail(new Error('Computer says no')));
+  });
+
+  it('resolves timed out asynchronous functions', async () => {
+    const res = await timeoutGo(operations.successFn, { timeoutMs: 5 });
+    expect(res).toEqual(fail(new Error('Operation timed out after final retry')));
+  });
+});
+
+describe('basic retryTimeoutGo usage', () => {
+  const operations = {
+    successFn: () =>
+      new Promise((res) =>
+        setTimeout(() => {
+          res(2);
+        }, 20)
+      ),
+    errorFn: () => new Promise((_res, rej) => setTimeout(() => rej(new Error('Computer says no')), 20)),
+  };
+
+  it('resolves successful asynchronous functions', async () => {
+    const res = await retryTimeoutGo(operations.successFn, { timeoutMs: 50, retries: 3 });
+    expect(res).toEqual(success(2));
+  });
+
+  it('resolves unsuccessful asynchronous functions', async () => {
+    const res = await retryTimeoutGo(operations.errorFn, { timeoutMs: 50, retries: 3 });
+    expect(res).toEqual(fail(new Error('Computer says no')));
+  });
+
+  it('retries and resolves successful asynchronous functions', async () => {
+    jest
+      .spyOn(operations, 'successFn')
+      .mockRejectedValueOnce(new Error('Error 1'))
+      .mockRejectedValueOnce(new Error('Error 2'));
+
+    const res = await retryTimeoutGo(operations.successFn, { timeoutMs: 100, retries: 3 });
+    expect(operations.successFn).toHaveBeenCalledTimes(3);
+    expect(res).toEqual(success(2));
+  });
+
+  it('retries and resolves unsuccessful asynchronous functions', async () => {
+    jest
+      .spyOn(operations, 'errorFn')
+      .mockRejectedValueOnce(new Error('Error 1'))
+      .mockRejectedValueOnce(new Error('Error 2'));
+
+    const res = await retryTimeoutGo(operations.errorFn, { timeoutMs: 100, retries: 2 });
+    expect(operations.errorFn).toHaveBeenCalledTimes(3);
+    expect(res).toEqual(fail(new Error('Computer says no')));
+  });
+
+  it('retries and resolves successful asynchronous functions', async () => {
+    jest
+      .spyOn(operations, 'successFn')
+      .mockRejectedValueOnce(new Error('Error 1'))
+      .mockRejectedValueOnce(new Error('Error 2'))
+      .mockRejectedValueOnce(new Error('Computer says no'));
+
+    const res = await retryTimeoutGo(operations.successFn, { timeoutMs: 100, retries: 2 });
+    expect(operations.successFn).toHaveBeenCalledTimes(3);
+    expect(res).toEqual(fail(new Error('Computer says no')));
+  });
```

Comment:
According to the name, this should end up being successful.

Reply:
:smile: Thanks for bearing with me. I've removed that test as the one [here](https://github.com/api3dao/promise-utils/pull/5/files#diff-258035e5968f6bf645400d417f310218d7d9a9a10606a3c34e7f55db193f58f3R163-R172) already tests that case.